src/bson.rs

```diff
@@ -58,18 +58,18 @@ pub enum Bson {
     JavaScriptCode(String),
     /// JavaScript code w/ scope
     JavaScriptCodeWithScope(JavaScriptCodeWithScope),
-    /// 32-bit integer
+    /// 32-bit signed integer
     I32(i32),
```

Comment:
These types are named `Int32`, `Int64` and `Uint64` in the spec, it seems inconsistent to use something different just for them (I imagine they are named this for parity with the built-in types, but that itself might be somewhat confusing)

Reply:
See my comment on `Bool` or the PR's description for the rationale behind the names for the integers.

`Timestamp` is a type we introduce ourselves, since it is a much easier API for users to use a struct instead of manually bit shifting an unsigned integer.